libbeat/outputs/amqp/client.go

```diff
@@ -487,10 +494,32 @@ func (c *client) prepareEvent(codec codec.Codec, incoming eventTracker, now time
 			ContentType:  c.contentType,
 			Body:         body,
 			MessageId:    messageID.String(),
+			Headers:      headers,
 		},
 	}, nil
 }
 
+// Extract headers from given event, return nil on failure
+func (c *client) getHeaders(content *beat.Event) amqp.Table {
+	if c.headersKey == "" {
+		return nil
+	}
+
+	c.logger.Debugf("Using headers key: %v", c.headersKey)
+
+	value, err := content.Fields.GetValue(c.headersKey)
+	if err != nil {
+		c.logger.Debugf("Error fetch headers with key %v: %v", c.headersKey, err)
+		return nil
+	}
+
+	if headers, ok := value.(common.MapStr); ok {
```

Comment:
If the value can't be cast to a MapStr then it should probably be considered an error to return rather than ignore

Reply:
Thanks @gwilym . I don't really expect any other type of values here because of the way events are parsed, which is why it's implemented this way. But I agree it's better to handle the error just in case. Will update!